src/lib/platform/exit_code.h

```diff
@@ -101,14 +102,16 @@ const char *otExitCodeToString(uint8_t aExitCode);
  * @param[in]   aExitCode   The exit code.
  *
  */
-#define VerifyOrDie(aCondition, aExitCode)                                                                   \
-    do                                                                                                       \
-    {                                                                                                        \
-        if (!(aCondition))                                                                                   \
-        {                                                                                                    \
-            otLogCritPlat("%s() at %s:%d: %s", __func__, __FILE__, __LINE__, otExitCodeToString(aExitCode)); \
-            exit(aExitCode);                                                                                 \
-        }                                                                                                    \
+#define VerifyOrDie(aCondition, aExitCode)                                                            \
+    do                                                                                                \
+    {                                                                                                 \
+        if (!(aCondition))                                                                            \
+        {                                                                                             \
+            otLogCritPlat("%s() at %s:%d: %s", __func__,                                              \
```

Comment:
Use a local variable to avoid multiple calls to `strrchr`?

```c++
        if (!(aCondition))                                                           \
        {                                                                            \
            const char *start = strrchr(__FILE__, '/');                              \
                                                                                     \
            otLogCritPlat("%s() at %s:%d: %s", __func__,                             \
                          (start != nullptr) ? start + 1 : __FILE__, __LINE__,       \
```